### PR TITLE
#68 - Introduce doc string format

### LINKED ISSUE #68: Introduce Doc string format
Make doc strings in one format.

### PR DESCRIPTION
**Pull request by miroslavpojer**
- Added new doc strings.
- Existing one formatted to follow same style.

Closes #68

### CODE REVIEW

**Comment by miroslavpojer** [comment]
Release notes
- Added new doc strings.
- Existing one formatted to follow same style.

File: main.py [hunk 1]
```diff
@@ -27,7 +33,7 @@
 
 def run():
     """
-    Runs the 'Release Notes Generator' GitHub Action.
+    The main function to run the Release Notes Generator.
```
Comment (MobiTikula) [inline main.py:36]: You are missing a @return value here.

File: release_notes_generator/action_inputs.py [hunk 2]
```diff
@@ -38,49 +42,114 @@
...
+    A class representing the inputs provided to the GH action.
+    The class is responsible for handling the inputs provided to the GH action.
+    """
+
     @staticmethod
     def get_github_repository() -> str:
+        """
+        Get the GitHub repository from the action inputs.
+
+        @return: The GitHub repository.
```
Comment (MobiTikula) [inline release_notes_generator/action_inputs.py:55]: Last time we were talking about documentation, we decided, that we won't write a `@return` for getters.

File: release_notes_generator/action_inputs.py [hunk 3]
```diff
@@ -38,49 +42,114 @@
 
 
 class ActionInputs:
+    """
+    A class representing the inputs provided to the GH action.
+    The class is responsible for handling the inputs provided to the GH action.
```
Comment (MobiTikula) [inline release_notes_generator/action_inputs.py:47]: You are often duplicating the information from module doc-string and class doc-string. `the inputs provided to the GH action.` is written in this class three times.

File: release_notes_generator/generator.py [hunk 4]
```diff
@@ -43,19 +54,22 @@ def __init__(self, github_instance: Github, custom_chapters: CustomChapters):
 
     @property
     def github_instance(self) -> Github:
+        """Getter for the GitHub instance."""
```
Comment (MobiTikula) [inline release_notes_generator/generator.py:57]: Here you work with getters differently than in Action Inputs.

File: release_notes_generator/model/service_chapters.py [hunk 5]
```diff
@@ -30,7 +34,8 @@
 # pylint: disable=too-many-instance-attributes
 class ServiceChapters(BaseChapters):
     """
-    A class used to represent the service chapters in the release notes.
+    A class representing the Service Chapters.
+    The class is responsible for representing the service chapters in the release notes.
```
Comment (MobiTikula) [inline release_notes_generator/model/service_chapters.py:38]: Maybe this is just my subjective (review) point of view. And I think I may do the same thing in my own work, I will look at that as well.
But information from the module doc-string is identical as in here in class doc. I would probably just write that the module contains a class called "abc". And the information about what is class representing and doing should be part of the class documentation.

File: release_notes_generator/record/record_factory.py [hunk 6]
```diff
@@ -41,17 +45,6 @@ class RecordFactory:
     def generate(
```
Comment (MobiTikula) [inline release_notes_generator/record/record_factory.py:45]: You are missing a method doc for this function.

File: release_notes_generator/utils/decorators.py [hunk 7]
```diff
@@ -41,9 +48,12 @@ def wrapped(*args, **kwargs) -> Optional[Any]:
...
-def safe_call_decorator(rate_limiter: GithubRateLimiter):
+def safe_call_decorator(rate_limiter: GithubRateLimiter) -> Callable:
     """
     Decorator factory to create a rate-limited safe call function.
+
+    @param rate_limiter: The rate limiter to use.
+    @return: The decorator.
     """
 
     def decorator(method: Callable) -> Callable:
```
Comment (MobiTikula) [inline release_notes_generator/utils/decorators.py:59]: I am not sure if we also want to have a doc for these decorator methods. It is up to debate. Same in GithubRateLimiter and Factory.

**Review by MobiTikula — COMMENTED** [review]
This PR has several suggestions, how to improve the code. And also one question up to debate.

File: release_notes_generator/action_inputs.py (hunk 2 above)
Comment (miroslavpojer) [reply release_notes_generator/action_inputs.py:55]: Yeah, my bad. I forgot about it in this big wave of changes.

File: release_notes_generator/action_inputs.py (hunk 3 above)
Comment (miroslavpojer) [reply release_notes_generator/action_inputs.py:47]: I have removed 1 of them. The rest I kept as module and class info.

File: release_notes_generator/model/service_chapters.py (hunk 5 above)
Comment (miroslavpojer) [reply release_notes_generator/model/service_chapters.py:38]: Yeah, I feel that module docs should describe it. Let's do it.

File: release_notes_generator/record/record_factory.py (hunk 6 above)
Comment (miroslavpojer) [reply release_notes_generator/record/record_factory.py:45]: Yes, I missed it. Fixed.

File: release_notes_generator/utils/decorators.py (hunk 7 above)
Comment (miroslavpojer) [reply release_notes_generator/utils/decorators.py:59]: Why not. It simply says, what is that ugly function? In the future, that can be expanded as needed.

**Review by MobiTikula — APPROVED** [review]
Approved.

**Pull request merged**